2021/TurcoZiccolella/app.py

```diff
@@ -0,0 +1,437 @@
+                            # create talker structure
+                            curr_talker = Talker(flow.dst_ip, flow.application_name, flow.src2dst_bytes,
+                                                 flow.dst2src_bytes, "ip"
+                                                 , flow.application_category_name, flow.application_is_guessed,
+                                                 flow.requested_server_name, flow.client_fingerprint,
+                                                 flow.server_fingerprint,
+                                                 flow.user_agent, flow.content_type)
+
+                            # create his RRD
+                            curr_talker.RRDcreate(rrd_step, hb, min_rrd_value, max_rrd_value, ts, creationtimestring)
+                            # insert in talkers
+                            talkers[flow.dst_ip] = curr_talker
+                            # new talker
+                            ntalkers += 1
+                        else:
+                            talkers[flow.dst_ip].update(ts, flow.src2dst_bytes, flow.dst2src_bytes)
+
+                ###################################################################################
+                elif filter_mode == "prot7":
+
+                    if flow.application_name not in talkers.keys():
+
+                        if ipaddress.ip_address(flow.src_ip).is_private:  # flow localhost -> talker
+                            # create talker structure
+                            curr_talker = Talker(flow.dst_ip, flow.application_name, flow.src2dst_bytes,
+                                                 flow.dst2src_bytes, "prot7"
+                                                 , flow.application_category_name, flow.application_is_guessed,
+                                                 flow.requested_server_name, flow.client_fingerprint,
+                                                 flow.server_fingerprint,
+                                                 flow.user_agent, flow.content_type)
+
+                        else:  # flow talker -> localhost
+                            # create talker structure
+                            curr_talker = Talker(flow.src_ip, flow.application_name, flow.dst2src_bytes,
+                                                 flow.src2dst_bytes, "prot7"
+                                                 , flow.application_category_name, flow.application_is_guessed,
+                                                 flow.requested_server_name, flow.client_fingerprint,
+                                                 flow.server_fingerprint,
+                                                 flow.user_agent, flow.content_type)
+
+                        # create his RRD
+                        curr_talker.RRDcreate(rrd_step, hb, min_rrd_value, max_rrd_value, ts, creationtimestring)
+                        # insert in talkers
+                        talkers[flow.application_name] = curr_talker
```

Comment:
Qui non gestite il caso pessimo andando a limitare il max numero di talker. Se uno vi fa un network scan, voi usate tutta la RAM, scoppiate e fate pure scoppiare il sistema. Dovete sempre, in tutti gli algoritmi, gestire i casi pessimi.